tensorflow-framework/src/main/java/org/tensorflow/framework/metrics/MeanRelativeError.java

```diff
@@ -0,0 +1,176 @@
+   *     will always produce the same random tensor for a given shape and data type.
+   * @param type the type for the variables and result
+   */
+  protected MeanRelativeError(Ops tf, String name, float[] normalizer, long seed, Class<T> type) {
+    this(tf, name, cast(tf, tf.constant(normalizer), type), seed, type);
+  }
+
+  /**
+   * Creates a MeanRelativeError metric using {@link Class#getSimpleName()} as the name
+   *
+   * @param tf the TensorFlow Ops
+   * @param normalizer The normalizer values with same shape as predictions.
+   * @param seed the seed for random number generation. An initializer created with a given seed
+   *     will always produce the same random tensor for a given shape and data type.
+   * @param type the type for the variables and result
+   */
+  protected MeanRelativeError(Ops tf, double[] normalizer, long seed, Class<T> type) {
+    this(tf, null, cast(tf, tf.constant(normalizer), type), seed, type);
+  }
+
+  /**
+   * Creates a MeanRelativeError metric
+   *
+   * @param tf the TensorFlow Ops
+   * @param name the name of the metric. If null, name defaults to {@link Class#getSimpleName()}.
+   * @param normalizer The normalizer values with same shape as predictions.
+   * @param seed the seed for random number generation. An initializer created with a given seed
+   *     will always produce the same random tensor for a given shape and data type.
+   * @param type the type for the variables and result
+   */
+  protected MeanRelativeError(Ops tf, String name, double[] normalizer, long seed, Class<T> type) {
+    this(tf, name, cast(tf, tf.constant(normalizer), type), seed, type);
+  }
+
+  /**
+   * Creates a MeanRelativeError metric using {@link Class#getSimpleName()} as the name
+   *
+   * @param tf the TensorFlow Ops
+   * @param normalizer The normalizer values with same shape as predictions.
+   * @param seed the seed for random number generation. An initializer created with a given seed
+   *     will always produce the same random tensor for a given shape and data type.
+   * @param type the type for the variables and result
+   */
+  protected MeanRelativeError(Ops tf, Operand<T> normalizer, long seed, Class<T> type) {
+    this(tf, null, normalizer, seed, type);
+  }
+
+  /**
+   * Creates a MeanRelativeError metric
+   *
+   * @param tf the TensorFlow ops
+   * @param name the name for this metric. If null, name defaults to {@link Class#getSimpleName()}.
+   * @param normalizer The normalizer values with same shape as predictions.
+   * @param seed the seed for random number generation. An initializer created with a given seed
+   *     will always produce the same random tensor for a given shape and data type.
+   * @param type the type for the variables and result
+   */
+  protected MeanRelativeError(
+      Ops tf, String name, Operand<T> normalizer, long seed, Class<T> type) {
+    super(tf, name, seed, type);
+    this.normalizer = normalizer;
+  }
+
+  /** {@inheritDoc} */
```

Comment:
Perhaps add class-specific documentation of the permitted shape relationship between `labels` and `predictions`?

Reply:
I think the only requirement is that the shape of the `normalizer` has to be the same shape as `predictions`, and that the `sampleWeights` has to be broadcastable to `labels`.

I have added specific method declarations instead of {@inheritdoc} on the updateStateList method. We should probably go back and do this for all the metrics.

Reply:
I have updated all the Javadoc on `updateStateList` methods for all Metrics